File: server/internal/model/customer.go
package model

import (
	"time"

	"gorm.io/gorm"
)

// CustomerStatus 表示客户状态。
type CustomerStatus int

const (
	// CustomerStatusEnabled 表示客户记录可正常使用。
	CustomerStatusEnabled CustomerStatus = 1
	// CustomerStatusDisabled 表示客户记录已停用。
	CustomerStatusDisabled CustomerStatus = 2
)

// Customer 是 CRM 客户档案表模型。
type Customer struct {
	ID           uint           `gorm:"primaryKey" json:"id"`
	Name         string         `gorm:"size:128;not null" json:"name"`
	ContactName  string         `gorm:"size:64;not null;default:''" json:"contact_name"`
	Phone        string         `gorm:"size:32;not null;default:''" json:"phone"`
	Level        string         `gorm:"size:32;not null;default:'';index" json:"level"`
	Source       string         `gorm:"size:32;not null;default:'';index" json:"source"`
	DepartmentID uint           `gorm:"not null;default:0;index" json:"department_id"`
	OwnerUserID  uint           `gorm:"not null;default:0;index" json:"owner_user_id"`
	Status       CustomerStatus `gorm:"type:smallint;not null;default:1;index" json:"status"`
	Remark       string         `gorm:"size:255;not null;default:''" json:"remark"`
	CreatedAt    time.Time      `json:"created_at"`
	UpdatedAt    time.Time      `json:"updated_at"`
	DeletedAt    gorm.DeletedAt `gorm:"index" json:"-"`
}

// TableName 固定 CRM 客户表名。
func (Customer) TableName() string {
	return "sys_customer"
}

File: admin/src/types/customer.ts
export const CustomerStatus = {
  Enabled: 1,
  Disabled: 2,
} as const

export type CustomerStatus = (typeof CustomerStatus)[keyof typeof CustomerStatus]

export interface CustomerItem {
  id: number
  name: string
  contact_name: string
  phone: string
  level: string
  source: string
  department_id: number
  department_name: string
  owner_user_id: number
  owner_username: string
  owner_nickname: string
  status: CustomerStatus
  remark: string
  created_at: string
  updated_at: string
}

export interface CustomerListQuery {
  page: number
  page_size: number
  keyword?: string
  level?: string
  source?: string
  status?: CustomerStatus | 0
}

export interface CustomerListResponse {
  items: CustomerItem[]
  total: number
  page: number
  page_size: number
}

export interface CreateCustomerPayload {
  name: string
  contact_name: string
  phone: string
  level: string
  source: string
  status: CustomerStatus
  remark: string
}

export type UpdateCustomerPayload = CreateCustomerPayload
